genai_stack/cli.py

```diff
@@ -195,6 +196,74 @@ def install(component, subcomponent, list_components, quickstart, config_file):
                 options = json.load(file)
                 Installer(component, subcomponent, options=options).install()
 
+@main.command()
+@click.option(
+    '--path', 
+    required=True, 
+    help="""Provide a directory path to setup the server files and default 
+        configurations files."""
+)
+@click.option('--host', default="127.0.0.1", help="Provide a host. eg: 127.0.0.1")
+@click.option(
+    '--port', 
+    default="8080", 
+    help="Provide a port number on which you want a server to run. eg: 8000"
+)
+def setup_server(path, host, port):
+
+    from mako.template import Template
+
+    # Load the Mako template
+    template_path = os.path.dirname(__file__)
+
+    main_template = Template(
+        filename=os.path.join(template_path, "templates/main.py.mako")
```

Comment:
use `pathlib` instead of `os` module. Its better at handling paths in different OS platforms.